### PR TITLE
Fix persistent connection request handling

### PR DESCRIPTION
**Pull request by deniskamazur**

### CODE REVIEW

File: persistent_stream.go [hunk 1]
```diff
@@ -45,67 +45,63 @@ func (d *Daemon) handlePersistentConn(r ggio.Reader, unsafeW ggio.WriteCloser) {
...
+	switch req.Message.(type) {
+	case *pb.PersistentConnectionRequest_AddUnaryHandler:
+		resp := d.doAddUnaryHandler(w, callID, req.GetAddUnaryHandler())
+
+		d.mx.Lock()
+		if _, ok := resp.Message.(*pb.PersistentConnectionResponse_DaemonError); !ok {
+			*streamHandlers = append(
+				*streamHandlers,
+				*req.GetAddUnaryHandler().Proto,
+			)
```
Comment (borzunov) [inline persistent_stream.go:68]: `append` is [not thread-safe](https://stackoverflow.com/questions/44152988/append-not-thread-safe), let's use a mutex.

**Review by borzunov — CHANGES_REQUESTED** [review]
Thanks for the fix, I've left a comment above.

File: persistent_stream.go (hunk 1 above)
Comment (deniskamazur) [reply persistent_stream.go:68]: what's wrong with the mutex on line 63?

Comment (borzunov) [reply persistent_stream.go:68]: Missed it, thanks :)

**Review by borzunov — APPROVED** [review]

**Pull request merged**